libs/telemetry/src/filter.rs

```diff
@@ -0,0 +1,88 @@
+use std::{borrow::Cow, sync::LazyLock};
+
+use tracing_subscriber::{filter::filter_fn, layer::Filter, EnvFilter};
+
+pub static SHOW_ALL_TRACES: LazyLock<bool> = LazyLock::new(|| {
+    std::env::var("PRISMA_SHOW_ALL_TRACES")
+        .map(|enabled| enabled.eq_ignore_ascii_case("true"))
+        .unwrap_or(false)
+});
+
+pub fn events<S>() -> impl Filter<S> {
+    filter_fn(|meta| meta.is_event())
+}
+
+pub fn user_facing_spans<S>() -> impl Filter<S> {
+    filter_fn(|meta| {
+        if !meta.is_span() {
+            return false;
+        }
+        *SHOW_ALL_TRACES || meta.fields().iter().any(|f| f.name() == "user_facing")
+    })
+}
+
+pub enum QueryEngineLogLevel<'a> {
+    FromEnv,
+    Override(&'a str),
+}
+
+impl<'a> QueryEngineLogLevel<'a> {
+    fn level(self) -> Option<Cow<'a, str>> {
+        match self {
+            Self::FromEnv => std::env::var("QE_LOG_LEVEL").ok().map(<_>::into),
+            Self::Override(level) => Some(level.into()),
+        }
+    }
+}
+
+pub struct EnvFilterBuilder<'a> {
+    log_queries: bool,
+    log_level: QueryEngineLogLevel<'a>,
+}
+
+impl<'a> EnvFilterBuilder<'a> {
+    pub fn new() -> Self {
+        Self {
+            log_queries: false,
+            log_level: QueryEngineLogLevel::FromEnv,
+        }
+    }
+
+    pub fn log_queries(mut self, log_queries: bool) -> Self {
+        self.log_queries = log_queries;
+        self
+    }
+
+    pub fn with_log_level(mut self, level: &'a str) -> Self {
+        self.log_level = QueryEngineLogLevel::Override(level);
+        self
+    }
+
+    pub fn build(self) -> EnvFilter {
+        let level = self.log_level.level().unwrap_or("info".into());
+
+        let mut filter = EnvFilter::from_default_env()
+            .add_directive("h2=error".parse().unwrap())
+            .add_directive("hyper=error".parse().unwrap())
+            .add_directive("tower=error".parse().unwrap())
```

Comment:
ditto, the QE server doesn't use tower, it was probably used by OTLP exporter with `tonic`